src/memory.jl

```diff
@@ -145,6 +142,106 @@ function transfer end
                     ATTACH_HOST   = 0x02)
                     #ATTACH_SINGLE = 0x04) # Defined but not valid
 
+@enum(CUmem_hostalloc, default       = 0x00,
```

Comment:
So I know this is a little weird using CuArrays within CUDAdrv. Not sure how else to go about this, but it does make the handling less awkward than using buffer objects directly.